test/Sign.Core.Test/TestInfrastructure/TemporaryEnvironmentPath.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+namespace Sign.Core.Test
+{
+    internal sealed class TemporaryEnvironmentPath : IDisposable
+    {
+        private const string Name = "PATH";
+
+        private readonly string? _originalEnvironmentPath;
+
+        internal TemporaryEnvironmentPath(string path)
+        {
+            _originalEnvironmentPath = Environment.GetEnvironmentVariable(Name);
+
+            string paths = _originalEnvironmentPath ?? string.Empty;
+            string newPaths = string.Join(Path.PathSeparator.ToString(), paths, path);
+
+            Environment.SetEnvironmentVariable(Name, newPaths);
+        }
+
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable(Name, _originalEnvironmentPath);
+
+            GC.SuppressFinalize(this);
```

Comment:
I've always acted under the guidance from this old StackOverflow answer -- that `GC.SuppressFinalize` should only be used if there's a finalizer: https://stackoverflow.com/a/151244. Could you help me understand the need for this line?

I understand that you want to leave the process env in it's original state, but it seems like it will only ever be best effort (caller invokes `Dispose` as expected) or unnecessary (process terminates without `Dispose` being called in which case calling `SetEnvironmentVariable` isn't needed anyway since the default overload you're calling uses `EnvironmentVariableTarget.Process` and the process's copy of env will be discarded anyway).

Reply:
Because the class is `sealed`, it's not necessary.  It's also no harm leaving it in; however, I'll remove it.